reactDashboard/src/components/DataCharts/QuickSelectDateRange.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
+import {
+  ALL_DATES,
+  LAST_WEEK,
+  LAST_TWO_WEEKS,
+  LAST_MONTH,
+  LAST_THREE_MONTHS,
+} from "../DataCharts/DataChartsConsts";
+import { calculateDateRange } from "./DataChartsUtils";
+
+function QuickSelectDateRange({ maxDateRange, setDateRange }) {
+  if (maxDateRange === undefined) {
+    throw new Error("missing maxDateRange");
+  }
+  const handleClick = (newTimePeriod) => {
+    setDateRange(calculateDateRange(maxDateRange, newTimePeriod));
+  };
+
+  return (
+    <div className="quick-select-dates">
+      <Button
```

Comment:
there's a lot of replication of code here - you could create a function createButton(id, timePeriod, buttonText) to reduce that.

Reply:
moved to separate ticket